<a href="https://colab.research.google.com/github/duvanr338-wq/Alura_Store/blob/main/Alura_Store.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ** Challenge Alura_store**

##Importando datos

In [2]:
import pandas as pd

In [9]:
import pandas as pd

url = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_1%20.csv"
url2 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_2.csv"
url3 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_3.csv"
url4 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_4.csv"

tienda = pd.read_csv(url)
tienda2 = pd.read_csv(url2)
tienda3 = pd.read_csv(url3)
tienda4 = pd.read_csv(url4)

tienda.head()

,Producto,Categoría del Producto,Precio,Costo de envío,Fecha de Compra,Vendedor,Lugar de Compra,Calificación,Método de pago,Cantidad de cuotas,lat,lon
0,Asistente virtual,Electrónicos,164300.0,6900.0,16/01/2021,Pedro Gomez,Bogotá,4,Tarjeta de crédito,8,4.60971,-74.08175
1,Mesa de comedor,Muebles,192300.0,8400.0,18/05/2022,Beatriz Morales,Medellín,1,Tarjeta de crédito,4,6.25184,-75.56359
2,Juego de mesa,Juguetes,209600.0,15900.0,15/03/2021,Juan Fernandez,Cartagena,1,Tarjeta de crédito,1,10.39972,-75.51444
3,Microondas,Electrodomésticos,757500.0,41000.0,03/05/2022,Juan Fernandez,Cali,4,Nequi,1,3.43722,-76.52250
4,Silla de oficina,Muebles,335200.0,20200.0,07/11/2020,Maria Alfonso,Medellín,5,Nequi,1,6.25184,-75.56359


##**Analisis de facturación**

In [10]:
tienda['Total_Facturación'] = tienda['Precio'] * tienda['Cantidad de cuotas']
tienda2['Total_Facturación'] = tienda2['Precio'] * tienda2['Cantidad de cuotas']
tienda3['Total_Facturación'] = tienda3['Precio'] * tienda3['Cantidad de cuotas']
tienda4['Total_Facturación'] = tienda4['Precio'] * tienda4['Cantidad de cuotas']


In [11]:
print("Facturación total de Tienda 1:", tienda['Total_Facturación'].sum())
print("Facturación total de Tienda 2:", tienda2['Total_Facturación'].sum())
print("Facturación total de Tienda 3:", tienda3['Total_Facturación'].sum())
print("Facturación total de Tienda 4:", tienda4['Total_Facturación'].sum())


Facturación total de Tienda 1: 3422717700.0
Facturación total de Tienda 2: 3337189200.0
Facturación total de Tienda 3: 3243816900.0
Facturación total de Tienda 4: 3098827200.0


##2. Ventas por categoría


In [12]:
sales_by_category_tienda1 = tienda.groupby('Categoría del Producto')['Total_Facturación'].sum()
sales_by_category_tienda2 = tienda2.groupby('Categoría del Producto')['Total_Facturación'].sum()
sales_by_category_tienda3 = tienda3.groupby('Categoría del Producto')['Total_Facturación'].sum()
sales_by_category_tienda4 = tienda4.groupby('Categoría del Producto')['Total_Facturación'].sum()


In [13]:
print("Ventas por categoría - Tienda 1:")
display(sales_by_category_tienda1)
print("\nVentas por categoría - Tienda 2:")
display(sales_by_category_tienda2)
print("\nVentas por categoría - Tienda 3:")
display(sales_by_category_tienda3)
print("\nVentas por categoría - Tienda 4:")
display(sales_by_category_tienda4)


Ventas por categoría - Tienda 1:


,Total_Facturación
Categoría del Producto,
Artículos para el hogar,3.831680e+07
Deportes y diversión,1.268837e+08
Electrodomésticos,1.032376e+09
Electrónicos,1.290391e+09
Instrumentos musicales,2.769905e+08
Juguetes,4.906280e+07
Libros,2.576940e+07
Muebles,5.829282e+08



Ventas por categoría - Tienda 2:


,Total_Facturación
Categoría del Producto,
Artículos para el hogar,4.273970e+07
Deportes y diversión,8.166060e+07
Electrodomésticos,1.026226e+09
Electrónicos,1.286644e+09
Instrumentos musicales,2.944487e+08
Juguetes,4.672120e+07
Libros,3.141700e+07
Muebles,5.273321e+08



Ventas por categoría - Tienda 3:


,Total_Facturación
Categoría del Producto,
Artículos para el hogar,4.472460e+07
Deportes y diversión,1.233992e+08
Electrodomésticos,1.081381e+09
Electrónicos,1.101334e+09
Instrumentos musicales,2.503360e+08
Juguetes,6.119040e+07
Libros,2.635890e+07
Muebles,5.550925e+08



Ventas por categoría - Tienda 4:


,Total_Facturación
Categoría del Producto,
Artículos para el hogar,4.457930e+07
Deportes y diversión,1.000620e+08
Electrodomésticos,8.790896e+08
Electrónicos,1.211446e+09
Instrumentos musicales,2.228372e+08
Juguetes,5.464120e+07
Libros,2.954340e+07
Muebles,5.566280e+08


##**3.Calificación promedio de la tienda**

In [14]:
average_rating_tienda1 = tienda['Calificación'].mean()
average_rating_tienda2 = tienda2['Calificación'].mean()
average_rating_tienda3 = tienda3['Calificación'].mean()
average_rating_tienda4 = tienda4['Calificación'].mean()


In [15]:
print("Calificación promedio - Tienda 1:", average_rating_tienda1)
print("Calificación promedio - Tienda 2:", average_rating_tienda2)
print("Calificación promedio - Tienda 3:", average_rating_tienda3)
print("Calificación promedio - Tienda 4:", average_rating_tienda4)


Calificación promedio - Tienda 1: 3.976685036032217
Calificación promedio - Tienda 2: 4.037303942348453
Calificación promedio - Tienda 3: 4.048325561678677
Calificación promedio - Tienda 4: 3.9957591178965224


##**4. Productos más y menos vendidos**


In [16]:
sales_by_product_tienda1 = tienda.groupby('Producto')['Total_Facturación'].sum()
sales_by_product_tienda2 = tienda2.groupby('Producto')['Total_Facturación'].sum()
sales_by_product_tienda3 = tienda3.groupby('Producto')['Total_Facturación'].sum()
sales_by_product_tienda4 = tienda4.groupby('Producto')['Total_Facturación'].sum()


In [17]:
print("Productos más y menos vendidos - Tienda 1:")
print("Más vendidos:")
display(sales_by_product_tienda1.nlargest(5))
print("\nMenos vendidos:")
display(sales_by_product_tienda1.nsmallest(5))

print("\nProductos más y menos vendidos - Tienda 2:")
print("Más vendidos:")
display(sales_by_product_tienda2.nlargest(5))
print("\nMenos vendidos:")
display(sales_by_product_tienda2.nsmallest(5))

print("\nProductos más y menos vendidos - Tienda 3:")
print("Más vendidos:")
display(sales_by_product_tienda3.nlargest(5))
print("\nMenos vendidos:")
display(sales_by_product_tienda3.nsmallest(5))

print("\nProductos más y menos vendidos - Tienda 4:")
print("Más vendidos:")
display(sales_by_product_tienda4.nlargest(5))
print("\nMenos vendidos:")
display(sales_by_product_tienda4.nsmallest(5))

Productos más y menos vendidos - Tienda 1:
Más vendidos:


,Total_Facturación
Producto,
TV LED UHD 4K,451704100.0
Iphone 15,276559900.0
Refrigerador,263591900.0
Smart TV,230818300.0
Secadora de ropa,216848800.0



Menos vendidos:


,Total_Facturación
Producto,
Cubo mágico 8x8,1490800.0
Dinosaurio Rex,1642300.0
Cuerda para saltar,1718700.0
Set de vasos,3295900.0
Ajedrez de madera,3869200.0



Productos más y menos vendidos - Tienda 2:
Más vendidos:


,Total_Facturación
Producto,
Iphone 15,392525900.0
Refrigerador,298656900.0
TV LED UHD 4K,284081800.0
Tablet ABXY,196546100.0
Smart TV,183417300.0



Menos vendidos:


,Total_Facturación
Producto,
Cubo mágico 8x8,1831100.0
Cuerda para saltar,2199900.0
Dinosaurio Rex,2281600.0
Set de vasos,3362100.0
Ajedrez de madera,3748700.0



Productos más y menos vendidos - Tienda 3:
Más vendidos:


,Total_Facturación
Producto,
Refrigerador,350098600.0
Iphone 15,276143300.0
TV LED UHD 4K,217222800.0
Smart TV,206708600.0
Lavadora de ropa,191895500.0



Menos vendidos:


,Total_Facturación
Producto,
Cubo mágico 8x8,1749600.0
Dinosaurio Rex,2257500.0
Cuerda para saltar,2832900.0
Set de vasos,4052600.0
Dashboards con Power BI,4103900.0



Productos más y menos vendidos - Tienda 4:
Más vendidos:


,Total_Facturación
Producto,
TV LED UHD 4K,315348300.0
Iphone 15,303017500.0
Refrigerador,267359000.0
Smart TV,217070800.0
Lavadora de ropa,182124000.0



Menos vendidos:


,Total_Facturación
Producto,
Cuerda para saltar,1789800.0
Dinosaurio Rex,2019300.0
Cubo mágico 8x8,2188100.0
Ajedrez de madera,3107200.0
Set de vasos,3856100.0


##**5. Envío promedio por tienda**

In [18]:
average_shipping_tienda1 = tienda['Costo de envío'].mean()
average_shipping_tienda2 = tienda2['Costo de envío'].mean()
average_shipping_tienda3 = tienda3['Costo de envío'].mean()
average_shipping_tienda4 = tienda4['Costo de envío'].mean()


In [19]:
print("Envío promedio - Tienda 1:", average_shipping_tienda1)
print("Envío promedio - Tienda 2:", average_shipping_tienda2)
print("Envío promedio - Tienda 3:", average_shipping_tienda3)
print("Envío promedio - Tienda 4:", average_shipping_tienda4)


Envío promedio - Tienda 1: 26018.60958033065
Envío promedio - Tienda 2: 25216.235693090293
Envío promedio - Tienda 3: 24805.680373039424
Envío promedio - Tienda 4: 23459.457167090754


#**6. Informe Final**
Señor Juan, despues de analizar y estudiar los informes de sus cuatro tiendas se concluyo que:

La Tienda 1 presenta los mayores ingresos, con un total de $1.150.880.400 pesos, aunque registra la calificación más baja entre las cuatro tiendas, con un promedio de calificación de 3.98 el más bajo y con el costo de envío más elevado.

La Tienda 2 se encuentra en segundo lugar de ingresos con un total de $1.116.343.500 pesos, con un calificación promedio de 4.04 superior a la tienda 1, y el costo de envío es moderado.

La Tienda 3 ** posee la mejor calificación promedio **4.05, con ingresos intermedios ($1.098.019.600 pesos), y el costo del envio es el tercero más bajo de las cuatro tiendas.

La ** Tienda 4** presenta los ingresos más bajos ($1.038.375.700 pesos), la calificación se encuentra es la segunda más baja 4.00 y el costo de envío es el más bajo.

Tienda 1 = mayor ingreso, calificación intermedia.
Tienda 2 = ingresos, calificación y envío intermedios.
Tienda 3 = mejor calificación, ingresos y envío intermedios.
Tienda 4 = calificación intermedia, ingresos más bajos y envío m´s barato.
Con respecto al las acategorias más vendidas por cada tienda, las cuatrro tiendas tienen las mismas categorias, Inmuebles en primer lugar seguido por Electrónicos y Juguetes. las categorias menos vendidas fueron Libros,instrumentos musicales y Articulos para el hogar.

Señor juan con la información dada se hayo que la Tienda 4 tiene el más bajo desempeño entre las cuatro tinedas, por tanto entre las cuatro tiendas, la Tienda 4 es la más recomendable para ser vendida.

